cmd/flux/bootstrap_gitlab.go

```diff
@@ -246,6 +246,19 @@ func bootstrapGitLabCmdRun(cmd *cobra.Command, args []string) error {
 		RecurseSubmodules: bootstrapArgs.recurseSubmodules,
 	}
 
+	// Read PGP Key
+	var entityList openpgp.EntityList
+	if bootstrapArgs.gpgKeyRingPath != "" {
+		r, err := os.Open(bootstrapArgs.gpgKeyRingPath)
+		if err != nil {
+			return fmt.Errorf("unable to open GPG key ring: %w", err)
+		}
+		entityList, err = openpgp.ReadKeyRing(r)
+		if err != nil {
+			return err
+		}
+	}
```

Comment:
This seems to be the same across `bitbucket`, `git`, `github` and `gitlab`. Can we extract as a func and reuse across them?

Reply:
Good idea. Moved it into a shared function.